acq4/devices/FilterWheel/FilterWheel.py

```diff
@@ -0,0 +1,402 @@
+# -*- coding: utf-8 -*-
+from acq4.devices.OptomechDevice import *
+from acq4.drivers.ThorlabsFW102C import *
+from acq4.devices.FilterWheel.FilterWheelDevGui import FilterWheelDevGui
+from acq4.devices.FilterWheel.FilterWheelTaskTemplate import Ui_Form
+from acq4.devices.Microscope import Microscope
+from acq4.util.SequenceRunner import *
+from acq4.devices.Device import *
+from acq4.devices.Device import TaskGui
+from acq4.util.Mutex import Mutex
+from acq4.util.Thread import Thread
+import acq4.util.debug as debug
+import time
+
+class FilterWheel(Device, OptomechDevice):
+    """ Thorlabs motorized filter wheel (FW102C)
+    The Filter wheel device class adds control and display for the filter wheel device status.
+    
+    * Maintains list of the filters in the six filter-wheel positions with their description
+    * Support for selecting a filter wheel position
+    * Support for filter wheel implementation during task : specific filter wheel position during one task, different positions as task sequence
+    * Support to changing filter wheel speed and input/ouput modus 
+    
+    Configuration examples:
+    
+    FilterWheel:
+        driver: 'FilterWheel'
+        port: 4                         ## serial port connected to filter-wheel
+        baud: 115200
+        parentDevice: 'Microscope'
+        filters: # filters in slots
+            0:  # first slot
+                name: 'green_and_shortpass'
+                description: 'ET535/70m Chroma, FESH0700 ThorL'
+            1:  # second slot 
+                name: 'red'
+                description: 'ET630/75m-2p Chroma'
+            2: # third slot
+                name: 'shortpass'
+                description: 'shortpass'
+
+
+    """
+    
+    sigFilterWheelPositionChanged = QtCore.Signal(object)
+    sigFilterWheelSpeedChanged = QtCore.Signal(object)
+    sigFilterWheelTrigModeChanged = QtCore.Signal(object)
+    sigFilterWheelSensorModeChanged = QtCore.Signal(object)
+    
+    def __init__(self, dm, config, name):
+        
+        Device.__init__(self, dm, config, name)
+        OptomechDevice.__init__(self, dm, config, name)
+        
+        self.scopeDev = None
+        self.currentFilter = None
+        p = self
+        while p is not None:
+            p = p.parentDevice()
+            if isinstance(p, Microscope):
+                self.scopeDev = p
+                
+        self.port = config['port']-1  ## windows com ports start at COM1, pyserial ports start at 0
```

Comment:
Better to let the user specify port as a string like "com1" or "/dev/ttyS0". `SerialDevice` should take care of this for you.